### ISSUE TITLE
`document.documentElement.outerHTML` only return the `head`

### ISSUE THREAD

**Issue by SettingDust** [WebReflection/linkedom#147]
![图片](https://user-images.githubusercontent.com/19922286/176984083-cc5df525-52af-4f67-a493-064b83dd34e5.png)

Code here: https://github.com/ndaidong/article-parser/blob/main/src/utils/stripUnwantedTags.js#L19=
Can test with `parseWithHtml` tests

**Comment by WebReflection** [comment]
@SettingDust is this a barrier language or can you write down a simple code that fails expectations (also written down)?!?

This is now you *setting* the `outerHTML` which I am not sure I care at all on the `html` level so this is a great candidate for a *won't fix* unless you explain to me what is it that you are trying to do ... second issue opened and I still don't understand what you are after.

Please explain with *more words* ... a code example in here, and feel free to use Google translator if that's the problem.

**Comment by WebReflection** [comment]
The example **sets** the `outerHTML` ... have you tried using the API as it's meant to be used instead? That means you pass the HTML to the `DOMParser`, not as `outerHTML` ... can you please use the API as shown in the README or the docs and tell me what is your expectation and what happens instead?

And once again, the `document.toString()` will never be the same of `documentElement.puterHTML` because the document includes the `<!doctype html>` part, when present.

**Comment by SettingDust** [comment]
```
assert(
  parseHTML('<head><title>test</title></head><body><div>asdf</div></body>').document.documentElement.outerHTML,
  '<head><title>test</title></head><body><div>asdf</div></body>'
);
```
will return
```
 Expected: <head><title>test</title></head><body><div>asdf</div></body>
 Got instead: <head><title>test</title></head>
```

**Comment by SettingDust** [comment]
MB, I didn't clarified. SRY

**Comment by SettingDust** [comment]
Thx a lot for patience

**Comment by WebReflection** [comment]
@SettingDust it's OK *but* ... `outerHTML` means `outerHTML` ... a `<html>` element with `outerHTML` expects `<html>...</html>` too, not just its content, 'cause that's `innerHTML`, not `outerHTML` ... this is why these issues you opened got me confused every single time ... if you set `outerHTML` of a `body`you need to pass the `<body>` in it too ... maybe doing that way it works too? I don't know, but I hope these exchanges helped 👋

**Comment by SettingDust** [comment]
The input of my use case isn't controlled. My program extract the article part of a page and input to `linkedom` for post process like remove unwanted tags. 
Maybe I should hold the element and operate the dom tree directly instead of parse the processed html.

**Comment by WebReflection** [comment]
@SettingDust yeah ... you are using the wrong tool for the job then, sorry .. this is not a browser, not JSDOM, nothing that will ever care about your use case so this is a won't fix indeed.
